src/main/java/software/amazon/cloudformation/resource/ResourceTypeSchema.java

```diff
@@ -148,6 +167,14 @@ public String getDescription() {
         return this.writeOnlyProperties.stream().map(JSONPointer::toString).collect(Collectors.toList());
     }
 
+    public Set<String> getHandlerPermissions(String action) {
```

Comment:
Return `null` is generally not a good practice as they need to check null for the result and it's easy to forget.

Let's provide other method ` boolean hasHandler(final String action)` so that they can use it to check if it has handler before to call those methods.

Reply:
cool i can do this. what should we do in cases where there isn't  a handler though? are you just saying to add the `hasHandler` and keep this the same?

Reply:
I'm kind of on the fence to enforce something like using `Optional<Set<String>>`, it could be overkill.  The same practice is Java `Map` interface. The `get` return `null` if doesn't exist or the value could be null. But need to at least provide an existence check method.

Let's keep it this way and provide an extra `hasHandler` method.